newrelic/hooks/external_botocore.py

```diff
@@ -40,43 +56,232 @@ def extractor_string(*args, **kwargs):
     return extractor_list
 
 
+def check_rate_limit_header(response_headers, header_name, is_int):
```

Comment:
We might want to extract this to a util file but either way let's make it match the implementation of openai:
https://github.com/newrelic/newrelic-python-agent/blob/develop-openai-instrumentation/newrelic/hooks/mlmodel_openai.py#L235C1-L248C18